### PR TITLE
New: Pagination Component

### LINKED ISSUE #697: Create a Pagination component
As of React Bootstrap [0.32](https://github.com/react-bootstrap/react-bootstrap/blob/master/CHANGELOG.md#v0320) the Pagination component has had a drastic refactor. This means we are unable to upgrade React Bootstrap without either releasing a breaking change into Adslot UI or to create a Pagination component internally.

## Types of issue

- [ ] Bug (High impact or critical to a production build)
- [ ] Bug (Low impact or not working as expected)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Something else (idea, build tool, documentation improvement)

## Description

To continue using React Bootstrap we need to export a pagination component

## Expected Behaviour and Accepted Behaviour

- [ ] Should provide a non-breaking upgrade path for the Pagination component
- [ ] Should mimic existing functionality

## Possible Solution

Create a new Pagination Component under `third-party/bootstrap/Pagination` that maps expected React Bootstrap Pagination v31 prop types to v32 components. Supporting existing functionality and behaviour.

### PR DESCRIPTION
**Pull request by omgaz**
## Motivation and Background Context

Resolves #697 

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## Screenshots (if appropriate):
after

## Check-list:

- [x] I have read the [Contributing](CONTRIBUTING.md) document.
- [x] I've thought about and labelled my PR/commit message appropriately.
- [ ] If this PR introduces breaking changes I've described the impact and migration path for existing applications.
- [x] CI is green (coverage, linting, tests).
- [x] I have updated the documentation accordingly.
- [ ] I've two LGTMs/Approvals.
- [ ] I've fixed or replied to all my code-review comments.
- [ ] I've manually tested with a buddy.
- [x] I've squashed my commits into one.

### CODE REVIEW

File: src/components/third-party/bootstrap/Pagination/index.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+import BootstrapPagination from 'react-bootstrap/lib/Pagination';
+
+class Pagination extends React.PureComponent {
+  render() {
+    const isFirst = this.props.activePage === 1;
+    const prevProps = isFirst
+      ? {
+          disabled: true,
+        }
+      : {
```
Comment (lteacher) [inline src/components/third-party/bootstrap/Pagination/index.jsx]: Ugh get #prettierRekt

File: src/components/third-party/bootstrap/Pagination/index.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+          onClick: () => this.props.onSelect(this.props.activePage - 1),
```
Comment (lteacher) [inline src/components/third-party/bootstrap/Pagination/index.jsx]: Cant you just rather control the disabled attribute?

```javascript
const prevProps = {
  disabled: isFirst,
  onClick: () => this.props.onSelect(this.props.activePage - 1),
};
```

File: src/components/third-party/bootstrap/Pagination/index.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+          const pageNumber = index + 1;
+          const isActive = pageNumber === this.props.activePage;
+          const itemProps = isActive
+            ? {
+                active: true,
+                disabled: true,
+              }
+            : {
+                onClick: () => this.props.onSelect(pageNumber),
+              };
```
Comment (lteacher) [inline src/components/third-party/bootstrap/Pagination/index.jsx]: Same as previous comment? isnt the `active` and `disabled` just controlled by `isActive`? e.g.

```javascript
{
  active: isActive,
  disabled: isActive,
  onClick: whatever...
}
```

File: src/components/third-party/bootstrap/Pagination/index.jsx (hunk 2 above)
Comment (lightbringer1991) [reply src/components/third-party/bootstrap/Pagination/index.jsx]: I'll investigate that

**Comment by omgaz** [comment]
screen shot 2018-05-29 at 09 25 52

getting this locally, re-installing my deps but hold off until it's good

**Comment by omgaz** [comment]
Yea was a local issue with my npm - one thing I noticed was the currently selected (active/disabled) state should be greyed out.

We have two examples in the platform that should probably be unified at some point

screen shot 2018-05-29 at 09 22 41
screen shot 2018-05-29 at 09 23 11

**Comment by lightbringer1991** [comment]
@omgaz the current implementation has active page as grey
screen shot 2018-05-30 at 10 50 12 am

File: src/components/third-party/bootstrap/Pagination/index.jsx [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+  activePage: PropTypes.number,
+  onSelect: PropTypes.func.isRequired,
+  next: PropTypes.bool,
+  prev: PropTypes.bool,
+};
+
+Pagination.defaultProps = {
+  activePage: 1,
+  next: true,
+  prev: true,
```
Comment (mdotwills) [inline src/components/third-party/bootstrap/Pagination/index.jsx]: I think defaults of `false` make sense for `next and `prev`?

File: docs/examples/PaginationExample.jsx [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+
+export const exampleProps = {
+  componentName: 'Pagination',
+  notes: (
+    <p>
+      See{' '}
+      <a href="https://getbootstrap.com/docs/3.3/components/#pagination" target="_blank" rel="noopener noreferrer">
+        Bootstrap documentation
+      </a>{' '}
+    </p>
```
Comment (omgaz) [inline docs/examples/PaginationExample.jsx]: Those spaces seem odd, but might have to do it: does this look better?

```html
<p>
  See <a
    href="https://getbootstrap.com/docs/3.3/components/#pagination"
    target="_blank"
    rel="noopener noreferrer"
  >Bootstrap documentation</a>
</p>
```

**Comment by lightbringer1991** [comment]
supersede https://github.com/Adslot/adslot-ui/pull/771

**Comment by vinteo** [comment]
like I mentioned in the other branch the button styles were broken too. but that will need to be another PR

**Pull request closed without merging**